src/suites/cts/validation/createView.spec.ts

```diff
@@ -0,0 +1,268 @@
+
+  const texture = t.createTexture({ arrayLayerCount: 1 });
+
+  const descriptor = t.getDescriptor({
+    dimension,
+    arrayLayerCount,
+    mipLevelCount,
+    baseMipLevel,
+  });
+
+  await t.expectValidationError(() => {
+    texture.createView(descriptor);
+  }, !success);
+}).params([
+  { arrayLayerCount: 1, success: true }, // it is OK to create a 2D texture view on a 2D texture
+  { arrayLayerCount: 2, success: false }, // it is an error to view a layer past the end of the texture
+  { dimension: '2d-array', arrayLayerCount: 1, success: true }, // it is OK to create a 1-layer 2D array texture view on a 2D texture
+  // baseMipLevel == k && mipLevelCount == 0 means to use levels k..end
+  { mipLevelCount: 0, baseMipLevel: 0, success: true },
+  { mipLevelCount: 0, baseMipLevel: 1, success: true },
+  { mipLevelCount: 0, baseMipLevel: MIP_LEVEL_COUNT - 1, success: true },
+  { mipLevelCount: 0, baseMipLevel: MIP_LEVEL_COUNT, success: false },
+  // it is an error to make the mip level out of range
+  { mipLevelCount: MIP_LEVEL_COUNT + 1, baseMipLevel: 0, success: false },
+  { mipLevelCount: MIP_LEVEL_COUNT, baseMipLevel: 1, success: false },
+  { mipLevelCount: 2, baseMipLevel: MIP_LEVEL_COUNT - 1, success: false },
+  { mipLevelCount: 1, baseMipLevel: MIP_LEVEL_COUNT, success: false },
+]);
+
+g.test('creating texture view on a 2D array texture', async t => {
+  const { dimension = '2d-array', arrayLayerCount, baseArrayLayer, success } = t.params;
+
+  const texture = t.createTexture({ arrayLayerCount: ARRAY_LAYER_COUNT_2D });
+
+  const descriptor = t.getDescriptor({
+    dimension,
+    arrayLayerCount,
+    baseArrayLayer,
+  });
+
+  await t.expectValidationError(() => {
+    texture.createView(descriptor);
+  }, !success);
+}).params([
+  { dimension: '2d', arrayLayerCount: 1, success: true }, // it is OK to create a 2D texture view on a 2D array texture
+  { arrayLayerCount: ARRAY_LAYER_COUNT_2D, success: true }, // it is OK to create a 2D array texture view on a 2D array texture
+  // baseArrayLayer == k && arrayLayerCount == 0 means to use layers k..end.
+  { arrayLayerCount: 0, baseArrayLayer: 0, success: true },
+  { arrayLayerCount: 0, baseArrayLayer: 1, success: true },
+  { arrayLayerCount: 0, baseArrayLayer: ARRAY_LAYER_COUNT_2D - 1, success: true },
+  { arrayLayerCount: 0, baseArrayLayer: ARRAY_LAYER_COUNT_2D, success: false },
+  // It is an error for the array layer range of the view to exceed that of the texture
+  { arrayLayerCount: ARRAY_LAYER_COUNT_2D + 1, baseArrayLayer: 0, success: false },
+  { arrayLayerCount: ARRAY_LAYER_COUNT_2D, baseArrayLayer: 1, success: false },
+  { arrayLayerCount: 2, baseArrayLayer: ARRAY_LAYER_COUNT_2D - 1, success: false },
+  { arrayLayerCount: 1, baseArrayLayer: ARRAY_LAYER_COUNT_2D, success: false },
+]);
+
+g.test(
+  'default values validates the same as specifying values for more than 1 array layer',
```

Comment:
"default values validates" - is this meant to read like this?

Reply:
Fixed.